packages/snowpack/src/commands/build.ts

```diff
@@ -285,16 +285,40 @@ export async function command(commandOptions: CommandOptions) {
     isHmrEnabled: false,
   });
 
-  if (config.buildOptions.minify) {
+  // note: the following logic is a little awkward, but we want to use the same
+  // esbuild service for minification AND/OR source maps
+  if (config.buildOptions.minify || config.buildOptions.sourceMaps) {
     const minifierStart = performance.now();
-    console.log(colors.yellow('! minifying javascript...'));
-    let minifierService = await esbuildStartService();
+    console.log(
+      colors.yellow(
+        config.buildOptions.minify ? '! minifying javascript...' : '! building source maps...',
+      ),
+    );
+    const minifierService = await esbuild.startService();
     const allJsFiles = glob.sync(path.join(buildDirectoryLoc, '**/*.js'));
-    for (const jsFile of allJsFiles) {
-      const jsFileContents = await fs.readFile(jsFile, 'utf-8');
-      const {js} = await minifierService.transform(jsFileContents, {minify: true});
-      js && (await fs.writeFile(jsFile, js, 'utf-8'));
-    }
+    await Promise.all(
+      allJsFiles.map(async (jsFile) => {
```

Comment:
Not sure if this is a significant perf improvement, but I think it’s probably a lot better than trying to write files in sequence.

Reply:
I'd still like to tackle the production minification use-case into a separate PR, if we can. This feels a bit like a hack on top of our current system (looks like it overwrites existing source maps, and relies on minification to strip the existing`sourceMapping` comment?). I'd rather tackle separately so that we can think through how this affects the existing source map story (and a general "combine multiple source maps" story).